Common/src/main/java/dk/s4_g1/common/services/IAPIService.java

```diff
@@ -0,0 +1,9 @@
+package dk.s4_g1.common.services;
+
+import dk.s4_g1.common.data.StatusCode;
+
+public interface IAPIService{
+    public StatusCode post(String endpoint, String data);
+    public StatusCode put(String endpoint, String data);
+    public StatusCode get(String endpoint, String data);
+}
```

Comment:
I think having `post`, `put` and `get` return a `StatusCode` class is not very extendable. There's more to a response than the status code. How about making a Response class which includes.

```java
public class Response{
    public final int statusCode // Or even better an enum
    public string response // json response, consider a HashMap
    public isClientError() {
        return this.statusCode > 300 and this.statusCode < 400
    }
}